python/ray/worker.py

```diff
@@ -804,7 +805,14 @@ def init(
     raylet_ip_address = node_ip_address
 
     if address:
-        redis_address, _, _ = services.validate_redis_address(address)
+        redis_address, _, redis_port = services.validate_redis_address(address)
+        if redis_port == 10001:
+            warnings.warn(
+                "Attempting to connect to redis at port 10001. 10001 is "
+                "the default port for the Ray client server. If you have "
+                "issues connecting, you may have meant to connect using Ray "
+                "client by prepending the address with \"ray://\", i.e. "
+                "`ray.init(\"ray://<head_node_ip>:10001\")`.", UserWarning)
```

Comment:
Rather than escaping with `\"`, it might be clearer to just use single quotes on the string and use double quotes inside.